internal/jsonrpc/jsonrpc.go

```diff
@@ -102,73 +101,50 @@ type DecodeError struct {
 	Content []byte // Content that couldn't be decoded.
 }
 
-func NewDecodeError(err error, content []byte) DecodeError {
-	return DecodeError{
+func NewDecodeError(err error, content []byte) *DecodeError {
+	return &DecodeError{
 		Err:     err,
 		Content: content,
 	}
 }
 
-func (e DecodeError) Error() string {
+func (e *DecodeError) Error() string {
 	return fmt.Sprintf("decode error: %s, content: %s", e.Err.Error(), string(e.Content))
 }
 
-func DecodeRequestBody(req *http.Request) (RequestBody, error) {
-	// No need to close the request body, the Server implementation will take care of it.
-	requestRawBytes, err := io.ReadAll(req.Body)
-
-	if err != nil {
-		return nil, NewDecodeError(err, requestRawBytes)
-	}
-
-	var body *SingleRequestBody
-
+func DecodeRequestBody(requestBodyRawBytes []byte) (RequestBody, error) {
```

Comment:
Moving the `io.ReadAll` call out of this function because errors reading data out of the stream were being conflated with real decoding errors.

Example from logs: `"error":"decode error: http2: server sent GOAWAY and closed the connection; LastStreamID=19999, ErrCode=NO_ERROR, debug=\"\",`

This is an error from `io.ReadAll`. It's an error on the connection level, not with the contents of the response.